### ISSUE TITLE
Made it work in 2020 for the commander pro

### ISSUE THREAD

**Issue by rodrigograca31** [benburkhart1/lighting-node-pro#6]
Steps:

1. delete `package-lock.json` and `node_modules` and run `npm install` again to install updated dependencies
2. change the following lines:

```
diff --git a/lib/commander.js b/lib/commander.js
index e41b6b7..6b2aec3 100644
--- a/lib/commander.js
+++ b/lib/commander.js
@@ -12,7 +12,7 @@ module.exports = class Commander {
         let devices = usb.getDeviceList();
 
         let foundDevices = devices.filter(d => {
-            return d.deviceDescriptor.idVendor === 0x1b1c && d.deviceDescriptor.idProduct === 0x0c0b;
+            return d.deviceDescriptor.idVendor === 0x1b1c && d.deviceDescriptor.idProduct === 0x0c10;
         });
 
         let devStrings = foundDevices.map(d => `(Bus ${d.busNumber} Address ${d.deviceAddress})`);
diff --git a/lib/lighting_node_pro.js b/lib/lighting_node_pro.js
index bcd17a2..cdee8f2 100644
--- a/lib/lighting_node_pro.js
+++ b/lib/lighting_node_pro.js
@@ -68,9 +68,10 @@ module.exports = class LightingNodePro {
         }
 
         console.log("Claiming interface.");
+
         this.iface.claim();
 
-        this.outEndpoint = this.iface.endpoint(1);
+        this.outEndpoint = this.iface.endpoints[1];
     }
```

**Comment by rodrigograca31** [comment]
Should I contribute the changes back or is this a dead project?
I loved that it used JS, my favorite language and that allowed me to play with it and make it work.

Im just not sure about that ID..... mine was that one.... I can still contribute the rest of the changes....

**Comment by rodrigograca31** [comment]
@benburkhart1 
Should I contribute the changes back or is this a dead project?
I loved that it used JS, my favorite language and that allowed me to play with it and make it work.

Im just not sure about that ID..... mine was that one.... I can still contribute the rest of the changes....

**Comment by rodrigograca31** [comment]
For those interested here is my fork:
https://github.com/rodrigograca31/lighting-node-pro

Also: had to run it using `sudo` ...

**Comment by rodrigograca31** [comment]
Im not sending a PR because Im not sure how that ID thing works. 
If only my ID was different or what...?

**Comment by benburkhart1** [comment]
Hey Rodrigo, your change looks good but you need to make the ID check an
OR, not replace what it was before.

**Comment by rodrigograca31** [comment]
So does that mean both IDs are valid!? Like the company produced devices with both IDs?
If thats the case yeah I can change it and send a PR back.

**Comment by mirsella** [comment]
+1 for deleting package-lock.json, when doing npm install it couldn't compile some usb package binary

**Comment by benburkhart1** [comment]
@rodrigograca31 Yes it means both IDs are valid, the old ones do not change, they are programmed into the hardware itself.

**Comment by benburkhart1** [comment]
So really the package-lock.json should be committed, I agree it needs to be ran with an updated version of nodejs so proper versions can be selected (it was written against an older version of node)

**Comment by mirsella** [comment]
@rodrigograca31
`return d.deviceDescriptor.idVendor === 0x1b1c && (d.deviceDescriptor.idProduct === 0x0c0b || d.deviceDescriptor.idProduct === 0x0c1a)` 
i've changed this and it work for my controller which isn't 0x0c0b but 0x0c1a, so it work for both normally

**Comment by rodrigograca31** [comment]
So it seems theres 3 ids as of now.... 🤔

`0x0c0b, 0x0c10, 0x0c1a`

Ok, I will send a PR in a few minutes....
